packages/aws-cdk-lib/aws-eks/lib/access-entry.ts

```diff
@@ -0,0 +1,340 @@
+        namespaces: options?.namespaces,
+      };
+    }
+    return new Import();
+  }
+  /**
+   * The scope of the access policy, which determines the level of access granted.
+   */
+  public readonly accessScope: AccessScope;
+
+  /**
+   * The access policy itself, which defines the specific permissions.
+   */
+  public readonly policy: string;
+
+  /**
+   * Constructs a new instance of the AccessPolicy class.
+   *
+   * @param {AccessPolicyProps} props - The properties for configuring the access policy.
+   */
+  constructor(props: AccessPolicyProps) {
+    this.accessScope = props.accessScope;
+    this.policy = props.policy.policyArn;
+  }
+}
+
+/**
+ * Represents the properties required to create an Amazon EKS access entry.
+ */
+export interface AccessEntryProps {
+  /**
+   * The name of the AccessEntry.
+   *
+   * @default - No access entry name is provided
+   */
+  readonly accessEntryName?: string;
+  /**
+   * The Amazon EKS cluster to which the access entry applies.
+   */
+  readonly cluster: ICluster;
+  /**
+   * The access policies that define the permissions and scope for the access entry.
+   */
+  readonly accessPolicies: IAccessPolicy[];
+  /**
+   * The Amazon Resource Name (ARN) of the principal (user or role) to associate the access entry with.
+   */
+  readonly principal: string;
+}
+
+/**
+ * Represents an access entry in an Amazon EKS cluster.
+ *
+ * An access entry defines the permissions and scope for a user or role to access an Amazon EKS cluster.
+ *
+ * @implements {IAccessEntry}
+ */
+export class AccessEntry extends Resource implements IAccessEntry {
+  /**
+   * Imports an `AccessEntry` from its attributes.
+   *
+   * @param scope - The parent construct.
+   * @param id - The ID of the imported construct.
+   * @param attrs - The attributes of the access entry to import.
+   * @returns The imported access entry.
+   */
+  public static fromAccessEntryAttributes(scope: Construct, id: string, attrs: AccessEntryAttributes): IAccessEntry {
+    class Import extends Resource implements IAccessEntry {
+      public readonly accessEntryName = attrs.accessEntryName;
+      public readonly accessEntryArn = attrs.accessEntryArn;
+
+    }
+    return new Import(scope, id);
+  }
+  /**
+   * The name of the access entry.
+   */
+  public readonly accessEntryName: string;
+  /**
+   * The Amazon Resource Name (ARN) of the access entry.
+   */
+  public readonly accessEntryArn: string;
+  private cluster: ICluster;
+  private principal: string;
+  private accessPolicies: IAccessPolicy[];
+
+  constructor(scope: Construct, id: string, props: AccessEntryProps ) {
+    super(scope, id);
+
+    this.cluster = props.cluster;
+    this.principal = props.principal;
+    this.accessPolicies = props.accessPolicies;
+
+    const resource = new CfnAccessEntry(this, 'Resource', {
+      clusterName: this.cluster.clusterName,
+      principalArn: this.principal,
+      accessPolicies: Lazy.any({
```

Comment:
For my learning purposes, the reason that we use `Lazy` here is because we have function below `addAccessPolicies` where users can add additional access policy outside of the constructor. So we make it lazy to make sure the conversion happens lazily?

Reply:
Yes Lazy is required if we need to evaluate and render an property on synthesize stage.

https://github.com/pahud/aws-cdk/blob/ceac30ac7355f712a1c96102464102afd4b358be/packages/aws-cdk-lib/aws-eks/lib/access-entry.ts#L314-L322

If we just set

```
accessPolicies: this.accessPolicies
```

then we won't be allowed to add more policies with the [addAccessPolicies](https://github.com/pahud/aws-cdk/blob/ceac30ac7355f712a1c96102464102afd4b358be/packages/aws-cdk-lib/aws-eks/lib/access-entry.ts#L336) method because that property would be determined on construct instantiation.

With Lazy, we could build the experience like

```ts
const entry = new AccessEntry(scope, id, {
  accessPolicies: myInitialAccessPolicy
})
entry.addAccessPolicies(...); // add additional policies on it
entry.addAccessPolicies(...); // add even more policies 
```

The Lazy function will render the final value for the L1 construct.

We have similar implementation in ECS [TaskDefinition](https://github.com/aws/aws-cdk/blob/205163fc0d2cac84d3d746a98c393e137f0e2388/packages/aws-cdk-lib/aws-ecs/lib/base/task-definition.ts#L485-L517) that renders the final container definition so our task definition allows us to keep adding containers like this and eventually render the final one for the L1.

```ts
task.addContainer(); // my first container def
task.addContainer(); // my second container def
task.addContainer(); // my 3rd container def
```